pkg/skaffold/build/buildpacks/lifecycle.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildpacks
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/docker/docker/api/types/mount"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+func (b *BuildpackBuilder) build(ctx context.Context, out io.Writer, workspace string, artifact *latest.BuildpackArtifact, tag string) (string, error) {
+	// To improve caching, we always build the image with [:latest] tag
+	// This way, the lifecycle is able to "bootstrap" from the previously built image.
+	// The image will then be tagged as usual with the tag provided by the tag policy.
+	parsed, err := docker.ParseReference(tag)
+	if err != nil {
+		return "", errors.Wrapf(err, "parsing tag %s", tag)
+	}
+	latest := parsed.BaseName + ":latest"
+
+	builderImage := artifact.Builder
```

Comment:
a buildpack doesn't always need to be a container image right? do we want to support supplying local buildpack directories too? i think working on the buildpack itself is a valid part of the development process.

Reply:
Yes, that's a valid use case. Not something I planned on doing as a first step, though.